### PR TITLE
Add possibility to load secret credentials from .env properties file

### PR DESCRIPTION
**Pull request by msimko81**

### CODE REVIEW

**Comment by msimko81** [comment]
Visual Studio changed EOL characters. Please review the pull request diff with the following link (ignores white space differencies):
https://github.com/Developer-Autodesk/design.automation.inventor-csharp-basics/pull/5/files?w=1

File: Solution/clientApp/Program.cs [hunk 1]
```diff
@@ -1,768 +1,821 @@
...
+            {
+                if(row.Trim().StartsWith("#"))
+                    continue;   //skip comments
+
+                var idx = row.IndexOf("=");
+                if (idx == -1)
+                    continue;
+
+                string key = row.Substring(0, idx).Trim();
+                string value = row.Length > idx ? row.Substring(idx + 1).Trim() : "";
```
Comment (erymski) [inline Solution/clientApp/Program.cs:644]: > `row.Length > idx`

always `true`?

File: Solution/clientApp/Program.cs [hunk 2]
```diff
@@ -1,768 +1,821 @@
...
+
+            if (!File.Exists(filename))
+                return properties;
+
+            foreach (var row in File.ReadAllLines(filename))
+            {
+                if(row.Trim().StartsWith("#"))
+                    continue;   //skip comments
+
+                var idx = row.IndexOf("=");
```
[the remark this answers is not in the document]
Comment (msimko81) [reply Solution/clientApp/Program.cs:639]: The original approach also skips commented out declarations, e.g.
`# CUSTOM_ENV = SOME_VALUE`

The second approach wouldn't skip this.

File: Solution/clientApp/Program.cs [hunk 3]
```diff
@@ -1,768 +1,821 @@
...
+
+            string consumerKeyPropFile;
+            string consumerSecretPropFile;
+
+            // read the security context from enviroment variables
+            string consumerKeySysEnv = System.Environment.GetEnvironmentVariable("FORGE_CLIENT_ID");
+            string consumerSecretSysEnv = System.Environment.GetEnvironmentVariable("FORGE_CLIENT_SECRET");
+
+            // read the security context from '.env' properties file located in user home dir
+            var properties = LoadProperties(System.Environment.GetEnvironmentVariable("userprofile") + "\\.env");
```
[the remark this answers is not in the document]
Comment (msimko81) [reply Solution/clientApp/Program.cs:600]: Actually my intention was that properties defined in `.env` file have higher priority than properties defined as system properties. 

But I'll change this.

**Comment by mivasi** [comment]
We will solve it later using asp.netcore way

**Pull request closed without merging**